File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    var sequence = [];
    for(i in progresses){
        var remain = (99 - progresses[i]) / speeds[i];
        sequence.push(Math.floor(remain)+1);
    }
    var comp = 0;
    var cnt = 1;
    var dic = {};
    sequence.forEach((e)=> {
        if(comp < e){
            comp = Math.max(comp,e);
            cnt = 1;
        }
        else{
            cnt += 1;
        }
        dic[comp] = cnt;
    })

    for (const key in dic) {
        answer.push(dic[key]);
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int add = 1;
    int max = (99 - progresses[0]) / speeds[0] + 1;
    for(int i = 1;i<progresses.size();i++){
        int check = (99 - progresses[i]) / speeds[i] + 1;
        if(max >= check){
            add++;
        }
        else{
            max = check;
            answer.push_back(add);
            add = 1;
        }
    }
    answer.push_back(add);
    return answer;
}